File: ExtensionBox-main/app/src/main/java/com/extensionbox/app/ui/ModuleRegistry.java
package com.extensionbox.app.ui;

public final class ModuleRegistry {

    public static final String[][] MODULES = {
        {"battery",    "🔋", "Battery",          "Current, power, temperature, health",     "true"},
        {"cpu_ram",    "🧠", "CPU & RAM",        "CPU usage, memory status",                "true"},
        {"screen",     "📱", "Screen Time",      "Screen on/off time, drain rates",         "true"},
        {"sleep",      "😴", "Deep Sleep",       "CPU sleep vs awake ratio",                "true"},
        {"network",    "📶", "Network Speed",    "Real-time download/upload speed",         "true"},
        {"data",       "📊", "Data Usage",       "Daily & monthly, WiFi & mobile",          "true"},
        {"unlock",     "🔓", "Unlock Counter",   "Daily unlocks, detox tracking",           "true"},
        {"storage",    "💾", "Storage",           "Internal storage usage",                  "false"},
        {"connection", "📡", "Connection Info",   "WiFi, cellular, VPN status",              "false"},
        {"uptime",     "⏱",  "Uptime",           "Device uptime since boot",                "false"},
        {"steps",      "👣", "Step Counter",      "Steps and distance",                      "false"},
        {"speedtest",  "🏎", "Speed Test",       "Periodic download/upload speed test",     "false"},
        {"fap",        "🍆", "Fap Counter",      "Self-monitoring counter & streak",        "false"},
    };

    public static String keyAt(int i)     { return MODULES[i][0]; }
    public static String emojiAt(int i)   { return MODULES[i][1]; }
    public static String nameAt(int i)    { return MODULES[i][2]; }
    public static String descAt(int i)    { return MODULES[i][3]; }
    public static boolean defAt(int i)    { return "true".equals(MODULES[i][4]); }
    public static int count()             { return MODULES.length; }

    public static String emojiFor(String key) {
        for (String[] m : MODULES) if (m[0].equals(key)) return m[1];
        return "?";
    }

    public static String nameFor(String key) {
        for (String[] m : MODULES) if (m[0].equals(key)) return m[2];
        return key;
    }
}

File: app/src/main/java/com/extensionbox/app/ui/ModuleRegistry.kt
package com.extensionbox.app.ui

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.*
import androidx.compose.ui.graphics.vector.ImageVector

data class ModuleDef(
    val key: String,
    val icon: ImageVector,
    val emoji: String, // Fallback for widgets (RemoteViews)
    val name: String,
    val description: String,
    val defaultEnabled: Boolean
)

object ModuleRegistry {

    private val MODULES = listOf(
        ModuleDef("battery", Icons.Rounded.BatteryChargingFull, "🔋", "Battery", "Current, power, temperature, health", true),
        ModuleDef("cpu_ram", Icons.Rounded.Memory, "🧠", "CPU & RAM", "CPU usage, memory status", true),
        ModuleDef("screen", Icons.Rounded.Smartphone, "📱", "Screen Time", "Screen on/off time, drain rates", true),
        ModuleDef("sleep", Icons.Rounded.Bedtime, "😴", "Deep Sleep", "CPU sleep vs awake ratio", true),
        ModuleDef("network", Icons.Rounded.NetworkCheck, "📶", "Network Speed", "Real-time download/upload speed", true),
        ModuleDef("data", Icons.Rounded.DataUsage, "📊", "Data Usage", "Daily & monthly, WiFi & mobile", true),
        ModuleDef("unlock", Icons.Rounded.LockOpen, "🔓", "Unlock Counter", "Daily unlocks, detox tracking", true),
        ModuleDef("storage", Icons.Rounded.Storage, "💾", "Storage", "Internal storage usage", false),
        ModuleDef("connection", Icons.Rounded.SettingsInputAntenna, "📡", "Connection Info", "WiFi, cellular, VPN status", false),
        ModuleDef("uptime", Icons.Rounded.History, "🕒", "Uptime", "Device uptime since boot", false),
        ModuleDef("steps", Icons.Rounded.DirectionsWalk, "👣", "Step Counter", "Steps and distance", false),
        ModuleDef("speedtest", Icons.Rounded.Speed, "🏎", "Speed Test", "Periodic download/upload speed test", false),
        ModuleDef("fap", Icons.Rounded.Favorite, "🍆", "Fap Counter", "Self-monitoring counter & streak", false)
    )

    fun keyAt(i: Int): String = MODULES[i].key
    fun iconAt(i: Int): ImageVector = MODULES[i].icon
    fun emojiAt(i: Int): String = MODULES[i].emoji
    fun nameAt(i: Int): String = MODULES[i].name
    fun descAt(i: Int): String = MODULES[i].description
    fun defAt(i: Int): Boolean = MODULES[i].defaultEnabled
    fun count(): Int = MODULES.size

    fun iconFor(key: String): ImageVector {
        return MODULES.find { it.key == key }?.icon ?: Icons.Rounded.Extension
    }

    fun emojiFor(key: String): String {
        return MODULES.find { it.key == key }?.emoji ?: "🧩"
    }

    fun nameFor(key: String): String {
        return MODULES.find { it.key == key }?.name ?: key
    }
}
